src/grpc/clients/ea_restaurant_java_etl_grpc_client.py

```diff
@@ -47,7 +65,54 @@ def insert_mongo_order_status_histories_from_python_etl(
                 mongoOrderStatusHistory=mongo_order_status_histories
             )
         )
-        metadata = (("authorization", access_token),)
-        return self.mongo_order_status_service_stub.insertMongoOrderStatusHistoriesFromPythonEtl(
-            mongo_order_status_histories_request, metadata=metadata
-        )
+
+        def run_grpc_call(metadata):
+
+            return self.mongo_order_status_service_stub.insertMongoOrderStatusHistoriesFromPythonEtl(
+                mongo_order_status_histories_request, metadata=metadata
+            )
+
+        response = self._run_with_oauth2(run_grpc_call)
+
+        return response.uuids
+
+    def _run_with_oauth2(self, grpc_call):
+
+        try:
+
+            if not self.access_token:
+                self._login_client(
+                    self.client_id,
+                    self.client_secret,
+                )
+
+            metadata = (
+                (
+                    "authorization",
+                    generate_bearer_token_from_access_token(self.access_token),
+                ),
+            )
+
+            return grpc_call(metadata)
+
+        except Exception as e:
+
+            if e.code().value[0] != StatusCode.PERMISSION_DENIED.value:
```

Comment:
Can we try to make it something more intuitive like:

```
if e.someParameter.error.code == StatusCode.PERMISSION_DENIED.value
```
?

I think using an array is difficult to follow `e.code().value[0]`.

I would also prefer a dictionary over the array in this case. unless you have an array of errors, then I would like to read:

```
first_error = e.errors[0]
```

Reply:
Done